packages/react/src/FormattingToolbar/components/DefaultButtons/ColorStyleButton.tsx

```diff
@@ -1,13 +1,36 @@
-import { useCallback } from "react";
+import { useCallback, useState } from "react";
 import { Menu } from "@mantine/core";
 import { BlockNoteEditor, BlockSchema } from "@blocknote/core";
 import { ToolbarButton } from "../../../SharedComponents/Toolbar/components/ToolbarButton";
 import { ColorIcon } from "../../../SharedComponents/ColorPicker/components/ColorIcon";
 import { ColorPicker } from "../../../SharedComponents/ColorPicker/components/ColorPicker";
+import { useEditorContentChange } from "../../../hooks/useEditorContentChange";
+import { useEditorSelectionChange } from "../../../hooks/useEditorSelectionChange";
 
 export const ColorStyleButton = <BSchema extends BlockSchema>(props: {
   editor: BlockNoteEditor<BSchema>;
 }) => {
+  const [currentTextColor, setCurrentTextColor] = useState<string>(
+    props.editor.getActiveStyles().textColor || "default"
+  );
+  const [currentBackgroundColor, setCurrentBackgroundColor] = useState<string>(
+    props.editor.getActiveStyles().backgroundColor || "default"
+  );
+
+  useEditorContentChange(props.editor, () => {
```

Comment:
@matthewlipski I would use a `forceUpdate` mechanism in `useEditorContentChange`. And maybe rename to `useUpdateOnContentChange` or sth. That way we can keep the old code and get rid of the separate state fields, which will be a lot cleaner I think